znai-diagrams/src/main/java/com/twosigma/znai/diagrams/graphviz/gen/GraphvizFromJsonGen.java

```diff
@@ -137,7 +157,17 @@ private String generateNodes(Collection<DiagramNode> nodes) {
     }
 
     private String generateNode(DiagramNode node) {
-        return node.getId() + " [label=\"" + generateNodeLabel(node) + "\"];";
+        return node.getId() + " [label=\"" + generateNodeLabel(node) + "\"" +
+                generateSizeAttributesIfRequired(node) + "];";
+    }
+
+    private String generateSizeAttributesIfRequired(DiagramNode node) {
+        if (!nodesConfig.isSizeDefined(node)) {
+            return "";
+        }
+
+        return " fixedsize=true" + (nodesConfig.isWidthDefined(node) ? " width=" + nodesConfig.getWidth(node) : "") +
```

Comment:
Let's use StringBuilder to make the code more readable.

    StringBuilder sb = new StringBuilder(" fixedsize=true");
    if (nodesConfig.isWidthDefined(node) {
        sb.append(" width=" + nodesConfig.getWidth(node));
    }
    if (nodesConfig.isHeightDefined(node) {
        sb.append(" height=" + nodesConfig.getHeight(node));
    }
    return sb.toString();

Reply:
done